base/file.jl

```diff
@@ -678,8 +685,41 @@ end
 # deprecated internal function used by some packages
 temp_cleanup_purge(; force=false) = force ? temp_cleanup_purge_all() : @lock TEMP_CLEANUP_LOCK temp_cleanup_purge_prelocked(false)
 
+function temp_cleanup_postprocess(cleanup_dirs)
+    if !isempty(cleanup_dirs)
+        rmcmd = """
+        eof(stdin)
+        for path in eachline(stdin)
+            try
+                rm(path, force=true, recursive=true)
+            catch ex
+                @warn "Failed to clean up temporary path \$(repr(path))\n\$ex" _group=:file
+            end
+        end
+        """
+        cmd = Cmd(Base.cmd_gen(((Base.julia_cmd(),), ("--startup-file=no",), ("-e",), (rmcmd,))); ignorestatus = true, detach = true)
+        pw = Base.PipeEndpoint()
+        rd, wr = Base.link_pipe(true, true)
```

Comment:
You're leaking rd handle here. Rather than fix this up though, I'd prefer switching to the fully managed API. You're right I forgot rawhandle didn't return an fd. The function I was looking for was `Base._fd`